src/fixed.rs

```diff
@@ -1,99 +1,62 @@
-    /// performing a copy.
-    fn encode_fixed_light<'a>(&'a self) -> &'a [u8];
+    /// Returns the representation of [`FixedInt`] as [`Bytes`], the little-endian representation
+    /// of self in the stack.
+    fn encode_fixed_light(self) -> Self::Bytes;
 
     /// Helper: Encode the value and return a Vec.
     fn encode_fixed_vec(self) -> Vec<u8> {
-        let mut v = Vec::new();
-        v.resize(Self::required_space(), 0);
-        self.encode_fixed(&mut v[..]);
-        v
+        self.encode_fixed_light().as_ref().to_vec()
     }
+
     /// Helper: Decode the value from the Vec.
     fn decode_fixed_vec(v: &Vec<u8>) -> Self {
-        assert_eq!(v.len(), Self::required_space());
         Self::decode_fixed(&v[..])
     }
 
     /// integer-encoding-rs always emits and receives little-endian integers (converting implicitly
     /// on big-endian machines). If you receive a big-endian integer, and would like it to be
     /// treated correctly, use this helper method to convert between endiannesses.
-    fn switch_endianness(self) -> Self {
-        // Switch to intrinsic bswap when out of nightly.
-        unsafe {
-            let sl = std::slice::from_raw_parts_mut(transmute::<&Self, *mut u8>(&self), Self::REQUIRED_SPACE);
-            sl.reverse();
-            *transmute::<*const u8, &Self>(sl.as_ptr())
-        }
-    }
+    fn switch_endianness(self) -> Self;
 }
 
 macro_rules! impl_fixedint {
     ($t:ty) => {
         impl FixedInt for $t {
-            const REQUIRED_SPACE: usize = size_of::<Self>();
-
-            fn required_space() -> usize {
-                Self::REQUIRED_SPACE
-            }
+            type Bytes = [u8; size_of::<$t>()];
 
-            fn encode_fixed_light<'a>(&'a self) -> &'a [u8] {
-                return unsafe {
-                    std::slice::from_raw_parts(
-                        transmute::<&$t, *const u8>(&self),
-                        Self::REQUIRED_SPACE,
-                    )
-                };
+            fn encode_fixed_light(self) -> Self::Bytes {
+                self.to_le_bytes()
             }
 
             fn encode_fixed(self, dst: &mut [u8]) {
-                assert_eq!(dst.len(), Self::REQUIRED_SPACE);
-
-                #[allow(unused_mut)]
-                let mut encoded =
-                    unsafe { &*(&self as *const $t as *const [u8; Self::REQUIRED_SPACE]) };
-
-                #[cfg(target_endian = "big")]
-                if Self::REQUIRED_SPACE > 1 {
-                    let mut encoded_rev = [0 as u8; Self::REQUIRED_SPACE];
-                    encoded_rev.copy_from_slice(encoded);
-                    encoded_rev.reverse();
-                    dst.clone_from_slice(&encoded_rev);
-                    return;
-                }
-
-                dst.clone_from_slice(encoded);
+                assert_eq!(dst.len(), size_of::<Self>());
+                dst.clone_from_slice(&self.to_le_bytes());
             }
 
             #[cfg(target_endian = "little")]
```

Comment:
I believe it was a mistake to introduce the target-dependent methods here. The byte order of some encoded data doesn't need to match the target endianness after all. (This is a note to myself, nothing you need to fix here.)